File: production-performance-monitor-backend-plus/backend/src/main/java/com/tasksyncpro/tasksyncpro/entity/Role.java
```java
package com.tasksyncpro.tasksyncpro.entity;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "roles")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Role {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    @Enumerated(EnumType.STRING)
    private RoleName name;

    @ManyToMany(mappedBy = "roles", fetch = FetchType.LAZY)
    private Set<User> users = new HashSet<>();

    public Role(RoleName name) {
        this.name = name;
    }

    public enum RoleName {
        USER,
        ADMIN
    }
}
```

File: fullstack-cms-system-advanced/backend/src/entities/Role.ts
import { Entity, PrimaryGeneratedColumn, Column, OneToMany } from 'typeorm';
import { User } from './User';

@Entity('roles')
export class Role {
  @PrimaryGeneratedColumn('uuid')
  id!: string;

  @Column({ unique: true, nullable: false })
  name!: string; // e.g., 'admin', 'editor', 'viewer'

  @Column({ nullable: true })
  description?: string;

  @OneToMany(() => User, (user) => user.role)
  users!: User[];
}